server/src/test/java/org/elasticsearch/action/index/IndexRequestTests.java

```diff
@@ -272,4 +281,102 @@ public void testRejectsEmptyStringPipeline() {
         assertThat(validate, notNullValue());
         assertThat(validate.getMessage(), containsString("pipeline cannot be an empty string"));
     }
+
+    public void testGetConcreteWriteIndex() {
```

Comment:
Can you add a test that uses epoch millis for the `@timestamp` instead of a string to ensure that it picks the right backing index also?

Reply:
I added a test for this and also added logic for this. The case of providing timestamp as a number representing mills since epoch failed. I fixes this via: d70aa28fc45bbf886b30584a753ccdc37f260343